moonshot/src/redteaming/attack/attack_module.py

```diff
@@ -27,18 +27,19 @@ class AttackModule:
         metric,prompt,prepared_prompt,system_prompt,predicted_result,duration,prompt_time)VALUES(?,?,?,?,?,?,?,?,?,?,?)
         """
 
-    def __init__(self, am_args: AttackModuleArguments):
-        self.id = am_args.name
-        self.description = "Overwrite this with your attack module description!"
-        self.name = am_args.name
-        self.connector_ids = am_args.connector_eps
-        self.prompt_templates = am_args.prompt_templates
-        self.prompt = am_args.prompt
-        self.system_prompt = am_args.system_prompt
-        self.metric_ids = am_args.metric_ids
-        self.context_strategy_info = am_args.context_strategy_info
-        self.db_instance = am_args.db_instance
-        self.params = am_args.params
+    def __init__(self, am_args: AttackModuleArguments | None = None):
+        if am_args:
```

Comment:
Why am_args will be None

Reply:
this is complex. will explain more later